sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -838,6 +843,31 @@ static ParsedNodes create(final SqlNode node) throws ValidationException
       return new ParsedNodes(explain, druidSqlInsert, query, ingestionGranularity);
     }
 
+    private static void validateSelectColumnForIngestion(SqlNode sqlNode) throws ValidationException
+    {
+      if (sqlNode instanceof SqlBasicCall) {
+        // If the sql node is a basic call, it could have select statements as operands.
+        // We need to recursivly check the operands for select statements.
+        SqlBasicCall sqlBasicCall = (SqlBasicCall) sqlNode;
+        for (SqlNode operand : sqlBasicCall.getOperandList()) {
+          validateSelectColumnForIngestion(operand);
+        }
+      } else if (sqlNode instanceof SqlSelect) {
+        // Validate that all columns which do some computation in the select statement are named.
+        // If the column is named, the top level operator should be "AS".
+        for (SqlNode column : ((SqlSelect) sqlNode).getSelectList().getList()) {
+          if (column instanceof SqlBasicCall) {
+            SqlBasicCall basicCall = (SqlBasicCall) column;
+            if (!"AS".equalsIgnoreCase(basicCall.getOperator().getName())) {
```

Comment:
Will this work if the alias is done without AS? Like `func(x) mycolumn`.

Reply:
Yes, Calcite implicitly uses the AS operator in the parse tree in such cases